src/components/Comments/ReplyCommentItem.js

```diff
@@ -141,6 +160,61 @@ const ReplyCommentItem = ({
     }
   };
 
+  const username = comment?.data?.anon_user_info_color_name
+    ? `Anonymous ${comment.data?.anon_user_info_emoji_name}`
+    : user?.data?.username;
+
+  const onPressDM = async () => {
+    try {
+      setLoading({...loading, loadingDm: true});
+      if (!comment?.data?.anon_user_info_color_name) {
+        const channelName = [username, profile?.myProfile?.username].join(',');
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [comment?.user?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(comment?.id, 'comment', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
+
+  const onPressDMAnon = async () => {
+    try {
+      setLoading({...loading, loadingDmAnon: true});
+      await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
+
+  const dataSheet = [
+    {
+      id: 1,
+      name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+      icon: <SendDMBlack />,
+      onPress: onPressDM
+    },
+    {
+      id: 2,
+      name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+      icon: <SendDMAnonBlock />,
+      onPress: onPressDMAnon
+    }
+  ];
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are two functions `onPressDM` and `onPressDMAnon` that have similar structure and do almost the same thing with minor differences. This is a violation of the DRY (Don't Repeat Yourself) principle. You can refactor these two functions into one function with an additional parameter to indicate whether the message is anonymous or not.

```diff
-  const onPressDM = async () => {
-    try {
-      setLoading({...loading, loadingDm: true});
-      if (!comment?.data?.anon_user_info_color_name) {
-        const channelName = [username, profile?.myProfile?.username].join(',');
-        const selectedUser = {
-          user: {
-            name: channelName,
-            image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
-          }
-        };
-        const members = [comment?.user?.id, profile.myProfile.user_id];
-        await createSignChat(members, selectedUser);
-      } else {
-        await sendMessageDM(comment?.id, 'comment', 'SIGNED');
-      }
-    } catch (e) {
-      console.warn(e);
-    } finally {
-      refSheet.current.close();
-      setLoading({...loading, loadingDm: false});
-    }
-  };
-
-  const onPressDMAnon = async () => {
-    try {
-      setLoading({...loading, loadingDmAnon: true});
-      await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
-    } catch (e) {
-      console.warn(e);
-    } finally {
-      refSheet.current.close();
-      setLoading({...loading, loadingDmAnon: false});
-    }
-  };
+  const onPressDM = async (isAnonymous = false) => {
+    try {
+      const loadingKey = isAnonymous ? 'loadingDmAnon' : 'loadingDm';
+      setLoading({...loading, [loadingKey]: true});
+
+      if (!isAnonymous && !comment?.data?.anon_user_info_color_name) {
+          const channelName = [username, profile?.myProfile?.username].join(',');
+          const selectedUser = {
+            user: {
+              name: channelName,
+              image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+            }
+          };
+          const members = [comment?.user?.id, profile.myProfile.user_id];
+          await createSignChat(members, selectedUser);
+      } else {
+          const messageType = isAnonymous ? 'ANONYMOUS' : 'SIGNED';
+          await sendMessageDM(comment?.id, 'comment', messageType);
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, [loadingKey]: false});
+    }
+  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->